src/model/MutableAreaDataSource.ts

```diff
@@ -301,7 +301,22 @@ export default class MutableAreaDataSource extends AreaDataSource {
         throw new Error('Area update error.  Reason: Updating leaf or boulder status of an area with subareas is not allowed.')
       }
 
-      if (areaName != null) area.set({ area_name: sanitizeStrict(areaName) })
+      if (areaName != null) {
+        const sanitizedName = sanitizeStrict(areaName)
+        area.set({ area_name: sanitizedName })
+
+        // change our pathTokens
+        const newPath = [...area.pathTokens]
+        newPath.pop()
+        newPath.push(sanitizedName)
+        area.set({ pathTokens: newPath })
+
+        // iterate over the DB id's in the child list and update pathTokens
+        for (const childId of area.children) {
+          await this.updatePathTokens(childId, sanitizedName)
```

Comment:
You should pass the `session` variable down to `updatePathTokens()` so that an exception will rollback everything.